sentinelhub/data_collections.py

```diff
@@ -0,0 +1,546 @@
+"""
+Module defining data collections
+"""
+import re
+import warnings
+from enum import Enum, EnumMeta
+from typing import Tuple
+from dataclasses import dataclass, asdict
+
+from aenum import extend_enum
+
+from .config import SHConfig
+from .constants import ServiceUrl
+from .exceptions import SHDeprecationWarning
+
+
+class _CollectionType:
+    """ Types of Sentinel Hub data collections
+    """
+    SENTINEL2 = 'Sentinel-2'
+    SENTINEL1 = 'Sentinel-1'
+    LANDSAT8 = 'Landsat 8'
+    MODIS = 'MODIS'
+    DEM = 'Mapzen DEM'
+    BYOC = 'BYOC'
+    BATCH = 'BATCH'
+    LANDSAT5 = 'Landsat 5'
+    LANDSAT7 = 'Landsat 7'
+    SENTINEL3 = 'Sentinel-3'
+    SENTINEL5P = 'Sentinel-5P'
+    ENVISAT_MERIS = 'Envisat Meris'
+
+
+class _SensorType:
+    """ Satellite sensors
+    """
+    # pylint: disable=invalid-name
+    MSI = 'MSI'
+    C_SAR = 'C-SAR'
+    OLCI = 'OLCI'
+    SLSTR = 'SLSTR'
+    TROPOMI = 'TROPOMI'
+
+
+class _ProcessingLevel:
+    """ Processing levels
+    """
+    # pylint: disable=invalid-name
+    L1B = 'L1B'
+    L1C = 'L1C'
+    L2A = 'L2A'
+    L3B = 'L3B'
+    L2 = 'L2'
+    GRD = 'GRD'
+
+
+class _SwathMode:
+    """ Swath modes for SAR sensors
+    """
+    # pylint: disable=invalid-name
+    IW = 'IW'
+    EW = 'EW'
+    SM = 'SM'
+    WV = 'WV'
+
+
+class _Polarization:
+    """ SAR polarizations
+    """
+    # pylint: disable=invalid-name
+    DV = 'DV'
+    DH = 'DH'
+    SV = 'SV'
+    SH = 'SH'
+    HH = 'HH'
+    HV = 'HV'
+    VV = 'VV'
+    VH = 'VH'
+
+
+class _Resolution:
+    """ Product resolution (specific to Sentinel-1 collections)
+    """
+    MEDIUM = 'MEDIUM'
+    HIGH = 'HIGH'
+
+
+class OrbitDirection:
+    """ Orbit directions
+    """
+    ASCENDING = 'ASCENDING'
+    DESCENDING = 'DESCENDING'
+    BOTH = 'BOTH'
+
+
+class _Bands:
+    """ Different collections of bands
+    """
+    SENTINEL2_L1C = ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08', 'B8A', 'B09', 'B10', 'B11', 'B12')
+    SENTINEL2_L2A = ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08', 'B8A', 'B09', 'B11', 'B12')
+    SENTINEL3_OLCI = tuple(f'B{str(index).zfill(2)}' for index in range(1, 22))
+    SENTINEL3_SLSTR = ('S1', 'S2', 'S3', 'S4', 'S5', 'S6', 'S7', 'S8', 'S9', 'F1', 'F2')
+    LANDSAT8 = ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08', 'B09', 'B10', 'B11')
+    DEM = ('DEM',)
+    MODIS = ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07')
+    SENTINEL1_IW = ('VV', 'VH')
+    SENTINEL1_EW = ('HH', 'HV')
+    SENTINEL1_EW_SH = ('HH',)
+    SENTINEL5P = ('AER_AI_340_380', 'AER_AI_354_388', 'CLOUD_BASE_HEIGHT', 'CLOUD_BASE_PRESSURE', 'CLOUD_FRACTION',
+                  'CLOUD_OPTICAL_THICKNESS', 'CLOUD_TOP_HEIGHT', 'CLOUD_TOP_PRESSURE', 'CO', 'HCHO', 'NO2', 'O3',
+                  'SO2', 'CH4')
+
+
+class _DataCollectionMeta(EnumMeta):
+    """ Meta class that builds DataCollection class enums
+    """
+    def __call__(cls, value, *args, **kwargs):
```

Comment:
This method made sense when we've only refactoring DataSource. But with the changes to DataCollection I don't see this method as needed.

Reply:
I agree. But I left it only for batchward-compatibility reasons. This way `DataSource('byoc-id')` still works but raises a deprecation warning.